Documentation/ApiOverview/BackendRouting/Index.rst

```diff
@@ -98,19 +104,50 @@ This technique should be used on all public routes (without session token) that
 internally redirect to a restricted route (having a session token). The goal is
 to protect and keep information about the current session token internal.
 
-The request sequence in the TYPO3 Core  looks like this:
+The request sequence in the TYPO3 Core looks like this:
 
 * HTTP request to `https://example.org/typo3/` having a valid user session
 * internally **public** backend route `/login` is processed
 * internally redirects to **restricted** backend route `/main` since an
   existing and valid backend user session was found
-  + HTTP redirect to `https://example.org/typo3/index.php?route=/main&token=...`
+  + HTTP redirect to `https://example.org/typo3/main?token=...`
   + exposing the token is mitigated with `referrer` route option mentioned above
 
 .. important::
 
    Please keep in mind these steps are part of a mitigation strategy, which requires
    to be aware of mentioned implications when implementing custom web applications.
+   
+   
+.. index:: Backend routing; Generating backend URLs
+   
+Generating backend URLs
+=======================
+
+Using the UriBuilder API, you can generate any kind of URL for the Backend, may it be
+a module, a typical route or an AJAX call. Therefore use either :php:`buildUriFromRoute()`
+or :php:`buildUriFromRoutePath()`. The UriBuilder then returns a PSR-7 conform Uri object
+that can be casted to string when needed. Furthermore does the UriBuilder automatically
+generate and apply the mentioned session token.
+
+.. code-block:: php
+
+   // Using a route identifier
+   $uriBuilder = GeneralUtility::makeInstance(UriBuilder::class);
```

Comment:
We should mention/link DI here IMO.

Reply:
What is "DI"?

Reply:
Dependency Injection: https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ApiOverview/DependencyInjection/Index.html

Reply:
This change just crept in here from rebasing. It was not part of this PR originally. So I would suggest you open an issue or PR about this and suggest a change.